crates/rooch-types/src/address.rs

```diff
@@ -821,7 +821,20 @@ impl TryFrom<MultiChainAddress> for BitcoinAddress {
 // Ref: https://github.com/nostr-protocol/nips/blob/master/19.md
 /// Nostr public key type
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub struct NostrPublicKey(pub XOnlyPublicKey);
+pub struct NostrPublicKey(XOnlyPublicKey);
+
+impl NostrPublicKey {
+    pub fn new(x_only_pk: XOnlyPublicKey) -> Self {
+        Self(x_only_pk)
+    }
+
+    /// Convert from the Nostr XOnlyPublicKey to Bitcoin address
+    pub fn to_bitcoin_address(&self) -> BitcoinAddress {
+        let x_only_pk_bytes = self.0.serialize().to_vec();
+        let bitcoin_address = BitcoinAddress::new(x_only_pk_bytes);
```

Comment:
Should not directly use BitcoinAddress::new to convert XOnlyPublicKey to Bitcoin address

Reply:
`NostrPublicKey` to `PublicKey::Secp256k1` to `BitcoinAddress`

Reply:
> `NostrPublicKey` to `PublicKey::Secp256k1` to `BitcoinAddress`

I referred to BIP-086 to generate a taproot address from `NostrPublicKey `.